lib/ansible/modules/storage/purestorage/purefa_dsrole.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2019, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefa_dsrole
+version_added: '2.8'
+short_description: Configure FlashArray Directory Service Roles
+description:
+- Set or erase directory services role configurations.
+- Only available for FlashArray running Purity 5.2.0 or higher
+author:
+- Simon Dodsley (@sdodsley)
+options:
```

Comment:
In a follow up PR could you please add `type:` details for all the options. This makes the online docs clearer. It's something we've been working in during Ansible 2.8 development.

A single PR which updates all you modules would be great, thanks in advance!